src/goto-instrument/cover_basic_blocks.cpp

```diff
@@ -19,6 +19,22 @@ Author: Peter Schrammel
 /// location of basic block, compress to ranges if applicable
 static void update_covered_lines(cover_basic_blockst::block_infot &block_info);
 
+/// If this block is a continuation of a previous block through unconditional
+/// forward gotos, return this blocks number.
+static optionalt<unsigned> continuation_of_block(
+  const goto_programt::const_targett &instruction,
+  std::map<goto_programt::const_targett, unsigned> &block_map)
```

Comment:
If this method was inside the class you could take advantage of the block_map typedef (it would be used twice then).